docs/building-a-form/common-patterns-for-building-forms.md

```diff
@@ -4,12 +4,295 @@ Some forms require custom validation, styles, or conditional information based o
 
 ### In this guide
```

Comment:
These headings aren't currently grouped in any intuitive fashion. I'm wondering about grouping them similarly to the short description (shortdesc) I wrote at the top:

- Creating custom validation for fields in a form
  - Writing custom field validation
  - Validating a field based on other fields in the same object
- Styling a form
  - Creating a block of text with no associated fields
  - Styling expanded or collapsed fields
- Displaying conditional fields and information in a form
  - Changing drop-down options based on data in another field
  - Conditionally hiding a group of fields
  - Conditionally including a page

Thoughts?

Reply:
I see what you mean, but I don't think there are enough things in this list to really require organizing and grouping them. The risk of creating groups is someone might look to the group names for what they're looking for and miss something that's relevant in another group. For example, there's a group on conditional stuff. But the topic `Validating a field based on other fields in the same object` under the first group is also related to conditional logic. If it were me I wouldn't put things into groups quite yet.

Reply:
Fair enough. I was anticipating the need for more of these given the conversations I've heard from you and @dmethvin-gov, but we can cross this bridge later on. 👍